Source/BugsnagNotifier.m

```diff
@@ -521,13 +521,13 @@ - (void)flushPendingReports {
 - (void)setupConnectivityListener {
     NSURL *url = self.configuration.notifyURL;
 
-    __weak id weakSelf = self;
-    self.networkReachable =
-        [[BSGConnectivity alloc] initWithURL:url
-                                 changeBlock:^(BSGConnectivity *connectivity) {
-                                   [weakSelf flushPendingReports];
-                                 }];
-    [self.networkReachable startWatchingConnectivity];
+    __weak typeof(self) weakSelf = self;
+    [BSGConnectivity monitorURL:url
+                  usingCallback:^(BOOL connected, NSString *connectionType) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
```

Comment:
A note (optionally with references, although Apple's relevant docs seem to have disappeared) might help explain the *why* of the `weakSelf/strongSelf` dance to developers not familiar with Objective C/ARC/Reference counting.  [This blog](https://dhoerl.wordpress.com/2013/04/23/i-finally-figured-out-weakself-and-strongself/), for instance, offers *an* explanation.